packaging/package/manifest.py

```diff
@@ -0,0 +1,144 @@
+"""
+Module for JAR manifest.
+"""
+
+import os
+
+from collections import OrderedDict
+from six import BytesIO
+
+
+class ManifestKeyException(Exception):
+
+    pass
+
+
+class ManifestSection(OrderedDict):
+
+    primary_key = "Name"
+
+    def __init__(self, name=None):
+        OrderedDict.__init__(self)
+        self[self.primary_key] = name
+
+    def store(self, stream, linesep=os.linesep):
+        """
+        Serialize this section and write it to a binary stream
+        """
+
+        for k, v in self.items():
+            write_key_val(stream, k, v, linesep)
+
+        stream.write(linesep.encode('utf-8'))
+
+    def get_data(self, linesep=os.linesep):
+        """
+        Serialize the section and return it as bytes
+        :return bytes
+        """
+
+        stream = BytesIO()
+        self.store(stream, linesep)
+        return stream.getvalue()
+
+
+class Manifest(ManifestSection):
+    """
+    Represents a Java Manifest as an ordered dictionary containing
+    the key:value pairs from the main section of the manifest, and
+    zero or more sub-dictionaries of key:value pairs representing the
+    sections following the main section. The sections are referenced
+    by the value of their 'Name' pair, which must be unique to the
+    Manifest as a whole.
+    """
+
+    primary_key = "Manifest-Version"
+
+    def __init__(self, version="1.0", linesep=None):
+        # can't use super, because we're a child of a non-object
+        ManifestSection.__init__(self, version)
+        self.sub_sections = OrderedDict([])
```

Comment:
I'm confused by this because I thought this class inherited from OrderedDict. Why does it also have one as a member?